test/javascripts/controllers/assign-user-test.js

```diff
@@ -0,0 +1,69 @@
+import EmberObject from "@ember/object";
+import pretender from "discourse/tests/helpers/create-pretender";
+import { discourseModule } from "discourse/tests/helpers/qunit-helpers";
+import { test } from "qunit";
+
+discourseModule("Unit | Controller | assign-user", function (hooks) {
+  hooks.beforeEach(function () {});
+
+  test("doesn't set suggestions and fails gracefully if controller is destroyed", function (assert) {
+    let resolveSuggestions;
+    pretender.get("/assign/suggestions", () => {
+      // eslint-disable-next-line no-restricted-globals
+      return new Promise((resolve) => {
```

Comment:
For now it's still recommended to use `import { Promise } from "rsvp";`

At some point we'll drop rsvp dependency but for now it's still standard in the codebase